### ISSUE TITLE
Child themes theme support do not override parent's theme.json settings

### ISSUE THREAD

**Issue by oandregal** [WordPress/gutenberg#35762]
### Description

Reported at https://github.com/WordPress/gutenberg/pull/35459#issuecomment-939989245

The settings for a theme are pulled from its `theme.json` and fall back to their theme supports if the equivalent `theme.json` setting is not present. When it comes to child themes we need to consider the whole chain of settings:

> parent theme supports => parent theme.json settings => child theme support => child theme.json settings

Options provided in the later stages of the chain should be able to override options provided before. For example, the child theme support settings should override the parent theme.json settings.

### Step-by-step reproduction instructions

- Create a child theme of TT1-blocks without `theme.json`.
- Add this code to the `functions.php` of the child:

```php
function twenty_twenty_one_child_setup() {
        remove_theme_support( 'experimental-link-color' );
        remove_theme_support( 'custom-line-height' );
}
add_action( 'after_setup_theme', 'twenty_twenty_one_child_setup', 15 );
```

The expected result is that, when you go to the editor, the link and line-height controls should not be present. Instead, what happens is that they are.

**Comment by Mamaduka** [comment]
@oandregal, I think this works as expected.

### Step-by-step reproduction instructions

1. Remove `settings.color.link` and `settings.typography.customLineHeight` from TT1 Blocks theme.json file.
2. Register theme support for TT1 Blocks, since it doesn't use them:

```php
add_theme_support( 'experimental-link-color' );
add_theme_support( 'custom-line-height' );
```
3. Create a child theme without a theme.json.
4. Remove theme support  in child using the following snippet:
```php
add_action( 'after_setup_theme', function() {
	remove_theme_support( 'experimental-link-color' );
	remove_theme_support( 'custom-line-height' );
}, 15 );
```

**Comment by Mamaduka** [comment]
Hi, @oandregal

Did you get a chance to test my latest findings?

**Comment by oandregal** [comment]
Oh, interesting. I've just checked [remove_theme_support](https://github.com/WordPress/wordpress-develop/blob/1d0e4441c108d854d6f1c7c8842eefb5140c4608/src/wp-includes/theme.php#L2920) and it just removes the property from the global. It's reasonable to think theme authors don't expect this to work because `remove_theme_support` never worked that way, instead we had to add things like `add_theme_support( 'disable-custom-colors' );`.

Thanks for following up on this, George!
